internal/store/installation.go

```diff
@@ -163,23 +163,72 @@ func (sqlStore *SQLStore) applyInstallationFilter(builder sq.SelectBuilder, filt
 	return builder
 }
 
-// GetInstallationsCount returns the number of installations filtered by the deletedat
-// field
-func (sqlStore *SQLStore) GetInstallationsCount(includeDeleted bool) (int, error) {
-	builder := sq.Select("COUNT(*) as InstallationsCount").From("Installation")
+// GetInstallationsCount returns the number of installations filtered by the
+// deleteAt field.
+func (sqlStore *SQLStore) GetInstallationsCount(includeDeleted bool) (int64, error) {
```

Comment:
I was thinking about doing something like this, but wasn't sure which one would be more performant. Do you happen to have any experience or data on if you think this would be better?

I ended up going with the current method simply because that is how we do it for our group logic as well. Happy to hear any additional thoughts on this though. Maybe good for a follow-up to refactor everything at once?

Reply:
Well the you would reduce the amount of memory (allocations) and the time given that you save 2 queries, you're already going through all the rows/stats from that database. I've made a small benchmark to compare the two options against my local PG, I'd expect more time improvement in production:

```
$ benchstat /tmp/old.txt /tmp/new.txt
name                                                    old time/op    new time/op    delta
GetInstallationsStatus/GetInstallationsStatus/1-16         289µs ± 1%      81µs ± 0%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/4-16         296µs ± 0%      86µs ± 5%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/16-16        314µs ± 1%     102µs ± 5%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/64-16        390µs ± 1%     161µs ± 3%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/256-16       638µs ± 5%     376µs ± 8%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/1024-16     1.68ms ± 1%    1.02ms ± 7%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/4096-16     5.50ms ± 1%    3.78ms ± 2%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/16384-16    21.0ms ± 1%    19.6ms ±34%   ~     (p=0.700 n=3+3)

name                                                    old alloc/op   new alloc/op   delta
GetInstallationsStatus/GetInstallationsStatus/1-16        10.7kB ± 0%     4.1kB ± 0%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/4-16        10.7kB ± 0%     4.1kB ± 0%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/16-16       10.7kB ± 0%     4.1kB ± 0%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/64-16       10.7kB ± 0%     4.1kB ± 0%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/256-16      10.7kB ± 0%     4.1kB ± 0%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/1024-16     10.7kB ± 0%     4.1kB ± 0%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/4096-16     10.7kB ± 0%     4.1kB ± 0%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/16384-16    10.7kB ± 0%     4.1kB ± 0%   ~     (p=0.100 n=3+3)

name                                                    old allocs/op  new allocs/op  delta
GetInstallationsStatus/GetInstallationsStatus/1-16           241 ± 0%        88 ± 0%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/4-16           242 ± 0%        89 ± 0%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/16-16          242 ± 0%        89 ± 0%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/64-16          243 ± 1%        90 ± 0%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/256-16         244 ± 0%        90 ± 0%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/1024-16        244 ± 0%        90 ± 0%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/4096-16        244 ± 0%        90 ± 0%   ~     (p=0.100 n=3+3)
GetInstallationsStatus/GetInstallationsStatus/16384-16       244 ± 0%        90 ± 0%   ~     (p=0.100 n=3+3)
```

As you can see at the time level is not a huge gain but you reduce allocations a lot

Reply:
Incredibly nice to see the data. Thanks for taking the time to run this. I definitely think this is worth the time to change so I will throw up this improvement shortly.